lib/rust/ensogl/core/src/display/shape/compound/rectangle.rs

```diff
@@ -52,21 +52,30 @@ pub mod shape {
             let canvas_width = canvas_width + &canvas_clip_width_diff.abs();
 
             // === Body ===
-            let inset2 = (&inset * 2.0).px();
+            let inset2 = (Max::max(inset.clone(), Var::from(0.0)) * 2.0).px();
             let width = &canvas_width - &inset2;
             let height = &canvas_height - &inset2;
-            let color = Var::<color::Rgba>::from(color);
             let body = Rect((&width, &height)).corners_radius(corner_radius.px());
-            let body = body.fill(color);
 
             // === Border ===
-            let padded_body = body.grow((inset - &border).px());
-            let border = padded_body.grow(border.px()) - padded_body;
-            let border_color = Var::<color::Rgba>::from(border_color);
-            let border = border.fill(border_color);
+            let border_center = &inset * border.negative() + &border * 0.5;
+            let border_thickness = border.abs() - f32::EPSILON;
```

Comment:
Is this so `0` is treated as negative? It could use a comment.

Reply:
Added a constant, and also made sure to actually what I planned correctly. Explanation is in the new comments.